cli/main.c

```diff
@@ -1434,6 +1434,88 @@ static int fw_info(int argc, char **argv)
 	return 0;
 }
 
+static char * part_to_string(int partition_id) {
+       switch(partition_id){
+
+               case 0: return "Partition Map 0";
+               case 1: return "Partition Map 1";
+               case 2: return "Key Manifest 0";
+               case 3: return "Key Manifest 1";
+               case 4: return "BL2 0";
+               case 5: return "BL2 1";
+               case 6: return "Cfg 0";
+               case 7: return "Cfg 1";
+               case 8: return "Main Fw 0";
+               case 9: return "Main FW 1";
+               default: return "Invalid Partition ID";
+       }
+}
+
+#define CMD_DESC_FW_META_INFO "return metadata information from the selected flash partition"
+
+static int fw_meta_info(int argc, char **argv)
+{
+       int ret, i, nr_info;
+       static struct {
+               struct switchtec_dev *dev;
+               int partition_id;
+       } cfg = {};
+
+       struct switchtec_fw_image_info inf;
+       struct switchtec_fw_metadata_gen4 *metadata;
+
+       const struct argconfig_options opts[] = {
+               DEVICE_OPTION,
+               {"flash_part", 'p', "", CFG_NONNEGATIVE, &cfg.partition_id, required_argument,
+                       "Flash Partition ID"},
```

Comment:
How does the user know which number for which partition ID? Why should the CLI be creating a new number for this? Other CLI commands use short strings to reference the partition type ("BOOT", "MAP", etc) not an opaque number.

Reply:
This partition ID is the partition number in the flash and device spec has the details. We cannot use the short strings as they only specify what type of partition it is and does not specify if we are looking at the active or redundant type of the partition.

Reply:
The partition ID can be inconsistent over generations in the firmware. Using short strings provides better compatibility.